devtools/etdump/buffer_data_sink.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <executorch/devtools/etdump/buffer_data_sink.h>
+#include <executorch/devtools/etdump/utils.h>
+
+using ::executorch::runtime::Error;
+using ::executorch::runtime::Result;
+
+namespace executorch {
+namespace etdump {
+
+Result<size_t> BufferDataSink::write(const void* ptr, size_t length) {
+  if (length == 0) {
+    return static_cast<size_t>(-1);
+  }
+  uint8_t* offset_ptr =
+      internal::align_pointer(debug_buffer_.data() + offset_, 64);
+
+  // Zero out the padding between data blobs.
+  size_t n_zero_pad = offset_ptr - debug_buffer_.data() - offset_;
+  memset(debug_buffer_.data() + offset_, 0, n_zero_pad);
+
+  offset_ = (offset_ptr - debug_buffer_.data()) + length;
```

Comment:
This seems correct but it's tough to follow. I suggest using temp pointers to simplify a lot of the arithmetic, like
```
uint8_t* base = debug_buffer_.data() + offset_;
uint8_t* new_base = internal::align_pointer(base, 64);
size_t padding = new_base - base;
memset(base, 0, padding);
offset_ += padding;
```